src/main/scala/sigmastate/serialization/ValueSerializer.scala

```diff
@@ -3,82 +3,64 @@ package sigmastate.serialization
 import sigmastate._
 import Values._
 
-import scala.util.Try
 import OpCodes._
 import sigmastate.SCollection.SByteArray
 import sigmastate.serialization.transformers._
-import sigmastate.serialization.trees.{QuadrupelSerializer, Relation2Serializer, Relation3Serializer}
+import sigmastate.serialization.trees.{QuadrupleSerializer, Relation2Serializer, Relation3Serializer}
 import sigmastate.utxo._
 import sigmastate.utils.Extensions._
-import Serializer.Consumed
 import org.ergoplatform._
 import sigmastate.lang.DeserializationSigmaBuilder
+import sigmastate.utils.{ByteReader, ByteWriter}
 
 
 trait ValueSerializer[V <: Value[SType]] extends SigmaSerializer[Value[SType], V] {
 
-  import ValueSerializer._
-
   override val companion = ValueSerializer
 
   /** Code of the corresponding tree node (Value.opCode) which is used to lookup this serizalizer
     * during deserialization. It is emitted immediately before the body of this node in serialized byte array. */
   val opCode: OpCode
 
-  override def toBytes(obj: V): Array[Byte] = serialize(obj)
-
-  override def parseBytes(bytes: Array[Byte]): Try[V] = Try {
-    deserialize(bytes).asInstanceOf[V]
-  }
 }
 
 object ValueSerializer extends SigmaSerializerCompanion[Value[SType]] {
   type Tag = OpCode
 
   val table: Map[OpCode, ValueSerializer[_ <: Value[SType]]] = Seq[ValueSerializer[_ <: Value[SType]]](
```

Comment:
@greenhat let's replace this map with Array. Where OpCode == index in the array.
Unused codes should be nulls.

Reply:
Done in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/191/commits/fdc5d1869584c1944e8e11121ae9cdb953721627